### PR TITLE
Removed ADATS in razor localization demo to use IAssemblyCatalog

### PR DESCRIPTION
**Pull request by jchannon**
Removes ADATS from razor localisation project as shown in #2221

### CODE REVIEW

File: samples/Nancy.Demo.Razor.Localization/CustomResourceAssemblyProvider.cs [hunk 1]
```diff
@@ -13,16 +11,22 @@
...
         public IEnumerable<Assembly> GetAssembliesToScan()
         {
             return (this.filteredAssemblies ?? (this.filteredAssemblies = GetFilteredAssemblies()));
         }
 
-        private static IEnumerable<Assembly> GetFilteredAssemblies()
+        private IEnumerable<Assembly> GetFilteredAssemblies()
         {
-            return AppDomainAssemblyTypeScanner.Assemblies.Where(x => !x.IsDynamic);
+            return this.assemblyCatalog.GetAssemblies(AssemblyResolveStrategies.All).Where(x => !x.IsDynamic);
```
Comment (khellang) [inline samples/Nancy.Demo.Razor.Localization/CustomResourceAssemblyProvider.cs]: Is `AssemblyResolveStrategies.All` needed here?

Comment (thecodejunkie) [inline samples/Nancy.Demo.Razor.Localization/CustomResourceAssemblyProvider.cs]: There should be an extension method that defaults to that

**Comment by jchannon** [comment]
Updated to use extension method and squashed it just for @khellang

File: samples/Nancy.Demo.Razor.Localization/CustomResourceAssemblyProvider.cs [hunk 2]
```diff
@@ -13,16 +11,22 @@
     /// </summary>
     public class CustomResourceAssemblyProvider : IResourceAssemblyProvider
     {
+        private readonly IAssemblyCatalog assemblyCatalog;
         private IEnumerable<Assembly> filteredAssemblies;
 
+        public CustomResourceAssemblyProvider(IAssemblyCatalog assemblyCatalog)
```
Comment (thecodejunkie) [inline samples/Nancy.Demo.Razor.Localization/CustomResourceAssemblyProvider.cs]: Missing constructor XML comment (including the parameter)

File: samples/Nancy.Demo.Razor.Localization/CustomResourceAssemblyProvider.cs [hunk 3]
```diff
@@ -13,16 +11,22 @@
...
         public IEnumerable<Assembly> GetAssembliesToScan()
         {
             return (this.filteredAssemblies ?? (this.filteredAssemblies = GetFilteredAssemblies()));
         }
 
-        private static IEnumerable<Assembly> GetFilteredAssemblies()
+        private IEnumerable<Assembly> GetFilteredAssemblies()
         {
-            return AppDomainAssemblyTypeScanner.Assemblies.Where(x => !x.IsDynamic);
+            return this.assemblyCatalog.GetAssemblies().Where(x => !x.IsDynamic);
```
Comment (khellang) [inline samples/Nancy.Demo.Razor.Localization/CustomResourceAssemblyProvider.cs]: Also, the `.Where(x => !x.IsDynamic)` call is redundant; https://github.com/NancyFx/Nancy/blob/fade252d9aa9c25fe190706fb7ae6b66b0da675c/src/Nancy/AppDomainAssemblyCatalog.cs#L30

File: samples/Nancy.Demo.Razor.Localization/CustomResourceAssemblyProvider.cs [hunk 4]
```diff
@@ -13,16 +11,22 @@
...
         public IEnumerable<Assembly> GetAssembliesToScan()
         {
             return (this.filteredAssemblies ?? (this.filteredAssemblies = GetFilteredAssemblies()));
         }
 
-        private static IEnumerable<Assembly> GetFilteredAssemblies()
+        private IEnumerable<Assembly> GetFilteredAssemblies()
         {
-            return AppDomainAssemblyTypeScanner.Assemblies.Where(x => !x.IsDynamic);
+            return this.assemblyCatalog.GetAssemblies();
```
Comment (khellang) [inline samples/Nancy.Demo.Razor.Localization/CustomResourceAssemblyProvider.cs]: Aaaand now we're at a point where I don't see the point in having this method at all :stuck_out_tongue_closed_eyes:

**Comment by jchannon** [comment]
I guess by default it does All, if you want to do something else you use your own?

File: samples/Nancy.Demo.Razor.Localization/CustomResourceAssemblyProvider.cs (hunk 4 above)
Comment (jchannon) [inline samples/Nancy.Demo.Razor.Localization/CustomResourceAssemblyProvider.cs]: I guess by default it does `All`, if you want to do something else you use your own?

Comment (thecodejunkie) [inline samples/Nancy.Demo.Razor.Localization/CustomResourceAssemblyProvider.cs]: Should just move the call to `GetAssemblies` to the end of line 24 and kill the method

**Comment by thecodejunkie** [comment]
In reality we could just pipe `this.assemblyCatalog.GetAssemblies()` straight through (it does caching internally per strategy) , but we can keep it like this because it makes no difference really

**Pull request merged**